bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsActions.java

```diff
@@ -110,25 +114,93 @@ private void setBypass(boolean from, int day, int month) {
+        }
+        return messages;
+    }
+
+    public static Map<String, Object> getErrorMessages(ThingActions actions) {
+        return ((HeliosEasyControlsActions) actions).getErrorMessages();
+    }
+
+    @RuleAction(label = "@text/action.getWarningMessages.label", description = "@text/action.getWarningMessages.description")
+    public @ActionOutput(name = "warningMessages", type = "java.util.ArrayList<String>") Map<String, Object> getWarningMessages() {
+        Map<String, Object> messages = new HashMap<>();
+        if (handler != null) {
+            messages.put("warningMessages", handler.getWarningMessages());
+        }
+        return messages;
+    }
+
+    public static Map<String, Object> getWarningMessages(ThingActions actions) {
+        return ((HeliosEasyControlsActions) actions).getWarningMessages();
+    }
+
+    @RuleAction(label = "@text/action.getInfoMessages.label", description = "@text/action.getInfoMessages.description")
+    public @ActionOutput(name = "infoMessages", type = "java.util.ArrayList<String>") Map<String, Object> getInfoMessages() {
+        Map<String, Object> messages = new HashMap<>();
+        if (handler != null) {
+            messages.put("infoMessages", handler.getInfoMessages());
+        }
+        return messages;
+    }
+
+    public static Map<String, Object> getInfoMessages(ThingActions actions) {
+        return ((HeliosEasyControlsActions) actions).getInfoMessages();
+    }
+
+    @RuleAction(label = "@text/action.getStatusMessages.label", description = "@text/action.getStatusMessages.description")
+    public @ActionOutput(name = "statusMessages", type = "java.util.ArrayList<String>") Map<String, Object> getStatusMessages() {
+        Map<String, Object> messages = new HashMap<>();
+        if (handler != null) {
+            messages.put("statusMessages", handler.getStatusMessages());
+        }
+        return messages;
+    }
+
+    public static Map<String, Object> getStatusMessages(ThingActions actions) {
+        return ((HeliosEasyControlsActions) actions).getStatusMessages();
+    }
+
+    @RuleAction(label = "@text/action.getMessages.label", description = "@text/action.getMessages.description")
+    public @ActionOutput(name = "errorMessages", type = "java.util.ArrayList<String>") @ActionOutput(name = "warningMessages", type = "java.util.ArrayList<String>") @ActionOutput(name = "infoMessages", type = "java.util.ArrayList<String>") @ActionOutput(name = "statusMessages", type = "java.util.ArrayList<String>") Map<String, Object> getMessages() {
+        Map<String, Object> messages = new HashMap<>();
+        if (handler != null) {
+            messages.put("errorMessages", handler.getErrorMessages());
+            messages.put("warningMessages", handler.getWarningMessages());
+            messages.put("infoMessages", handler.getInfoMessages());
+            messages.put("statusMessages", handler.getStatusMessages());
```

Comment:
You could store `handler` to a local variable and do the null check on that to get rid of the compiler warnings.